demos/python_demos/human_pose_estimation_3d_demo/modules/input_reader.py

```diff
@@ -0,0 +1,43 @@
+import cv2
+
+
+class ImageReader:
+    def __init__(self, file_names):
+        self.file_names = file_names
+        self.max_idx = len(file_names)
+
+    def __iter__(self):
+        self.idx = 0
+        return self
+
+    def __next__(self):
+        if self.idx == self.max_idx:
+            raise StopIteration
+        img = cv2.imread(self.file_names[self.idx], cv2.IMREAD_COLOR)
+        if img.size == 0:
+            raise IOError('Image {} cannot be read'.format(self.file_names[self.idx]))
+        self.idx = self.idx + 1
+        return img
+
+
+class VideoReader:
+    def __init__(self, file_name):
+        self.file_name = file_name
+        try:  # OpenCV needs int to read from webcam
+            self.file_name = int(file_name)
+        except ValueError:
+            pass
```

Comment:
You can update `self.file_name = file_name` in case of exception risen instead of unconditionally storing it in the beginning of `__init__`.